core/src/test/scala/org/broadinstitute/dsde/rawls/monitor/WorkspaceMigrationMonitorSpec.scala

```diff
@@ -209,15 +231,67 @@ class WorkspaceMigrationMonitorSpec
           tmpBucketName = GcsBucketName(sourceBucket).some
         )
 
-      val (job, writeAction) = WorkspaceMigrationMonitor.startBucketStorageTransferJob(
-            migration,
-            GcsBucketName(v1Workspace.bucketName),
-            migration.tmpBucketName.get,
-            GoogleProject("to-be-determined"),
-            stsMock
-          ).unsafeRunSync
+      val writeAction = WorkspaceMigrationMonitor.deleteTemporaryBucket(
+        migration,
+        new MockGoogleStorageService[IO] {
+          override def deleteBucket(googleProject: GoogleProject, bucketName: GcsBucketName, isRecursive: Boolean, bucketSourceOptions: List[Storage.BucketSourceOption], traceId: Option[TraceId], retryConfig: RetryConfig): fs2.Stream[IO, Boolean] =
+            fs2.Stream.emit(true)
+        }
+      ).unsafeRunSync
+
+      runAndWait(writeAction) shouldBe()
+
+      val tmpBucketDeleted = runAndWait(
+        WorkspaceMigrationMonitor.workspaceMigrations
+          .filter(_.id === migration.id)
+          .map(_.tmpBucketDeleted)
+          .result
+          .map(_.head)
+      )
+
+      tmpBucketDeleted shouldBe defined
+    }
+  }
+
+  val mockStsService = new MockGoogleStorageTransferService[IO] {
```

Comment:
I prefer this implementation to the individual overrides above, but it could still lead to boilerplate if we need to use this service elsewhere. Same question applies, do you feel as though the added clarity is worth the added boilerplate?